src/python_inspector/resolution.py

```diff
@@ -150,8 +212,76 @@ def get_requirements_for_package_from_pypi_simple(self, candidate):
                 if dep.scope == "install":
                     yield packaging.requirements.Requirement(str(dep.extracted_requirement))
 
-    def get_requirements_for_package_from_pypi_json_api(self, purl):
+        sdist_file = get_sdist_file(
+            repos=self.repos, candidate=candidate, python_version=python_version
+        )
+
+        if sdist_file:
+            setup_py_path = os.path.join(
+                utils_pypi.CACHE_THIRDPARTY_DIR,
+                "extracted_sdists",
+                sdist_file,
+                sdist_file,
+                "setup.py",
+            )
+            setup_cfg_path = os.path.join(
+                utils_pypi.CACHE_THIRDPARTY_DIR,
+                "extracted_sdists",
+                sdist_file,
+                sdist_file,
+                "setup.cfg",
+            )
+            pkg_info_path = os.path.join(
```

Comment:
With the design suggested above, you can just get the os.path.dirname here since the function would now return a full directory, one level below